src/agent/debuglet.ts

```diff
@@ -262,6 +262,24 @@ export class Debuglet extends EventEmitter {
         return;
       }
 
+      if (onGCP &&
+          (!that.config_.serviceContext ||
+           !that.config_.serviceContext.service)) {
```

Comment:
Ack. I think the behaviour of fetching the cluster name on an empty string sounds reasonable still, so I agree this is not an issue.